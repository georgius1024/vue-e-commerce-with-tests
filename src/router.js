import * as VueRouter from "vue-router";

import Home from "@/views/Home.vue";
import Category from "@/views/Category.vue";
import Product from "@/views/Product.vue";
import Checkout from "@/views/Checkout.vue";
import Error from "@/views/Error404.vue";
const routes = [
  { path: "/", name: "home", component: Home },
  { path: "/:category", name: "category", component: Category, props: true },
  { path: "/:category/:id", name: "product", component: Product, props: true },
  { path: "/checkout", name: "checkout", component: Checkout },
  { path: "/:pathMatch(.*)*", component: Error },
];
export default VueRouter.createRouter({
  history: VueRouter.createWebHashHistory(),
  routes,
});
