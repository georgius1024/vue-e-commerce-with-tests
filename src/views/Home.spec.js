import { describe, it, expect, vi } from "vitest";
import { mount, flushPromises } from "@vue/test-utils";
import component from "@/views/Home.vue";
import router from "@/router";

vi.mock("@/services/categories", () => ({
  default: vi.fn().mockResolvedValue(global.servicesMock),
}));

describe("Home page", () => {
  const buildWrapper = () => {
    return mount(component, {
      global: {
        plugins: [router],
      },
    });
  };
  it("mounts without error", async () => {
    const wrapper = buildWrapper();
    expect(wrapper.exists).toBeTruthy();
    expect(wrapper.html()).toMatchSnapshot();
  });
  it("use service to obtain data", async () => {
    const wrapper = buildWrapper();
    await flushPromises();
    expect(wrapper.html()).toMatchSnapshot();
    global.servicesMock.forEach(name => expect(wrapper.html()).toContain(name))
  });
});
