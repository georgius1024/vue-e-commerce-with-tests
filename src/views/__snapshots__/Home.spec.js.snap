// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Home page > mounts without error 1`] = `
"<div class=\\"container\\">
  <ol class=\\"breadcrumb\\">
    <li class=\\"breadcrumb-item active\\">Categories</li>
  </ol>
  <h1>Categories</h1>
  <ul class=\\"list-group\\"></ul>
</div>"
`;

exports[`Home page > use service to obtain data 1`] = `
"<div class=\\"container\\">
  <ol class=\\"breadcrumb\\">
    <li class=\\"breadcrumb-item active\\">Categories</li>
  </ol>
  <h1>Categories</h1>
  <ul class=\\"list-group\\">
    <li class=\\"list-group-item text-capitalize\\"><a href=\\"#/category%201\\" class=\\"\\">category 1</a></li>
    <li class=\\"list-group-item text-capitalize\\"><a href=\\"#/category%202\\" class=\\"\\">category 2</a></li>
  </ul>
</div>"
`;
